File: Labs/Lab-04-20231013/app/src/main/java/sap/escooters/domain_layer/DomainLayer.java
package sap.escooters.domain_layer;

import java.util.Optional;

import sap.layers.Layer;

public interface DomainLayer extends Layer {

	void addNewUser(String id, String name, String surname);
	Optional<User> getUser(String userId);

	void addNewEScooter(String id);
	Optional<EScooter> getEScooter(String id);
	
	String startNewRide(User user, EScooter escooter);
	Optional<Ride> getRide(String rideId);
}

File: Labs/Lab-05-20231016/app/src/main/kotlin/assigment03/escooters/domain_layer/DomainLayer.kt
package assigment03.escooters.domain_layer

import assigment03.layers.Layer

interface DomainLayer: Layer {
    fun addNewUser(id: String, name: String, surname: String)
    fun getUser(userId: String): User?
    fun addNewEScooter(id: String)
    fun getEScooter(id: String): EScooter?
    fun startNewRide(user: User, escooter: EScooter): String
    fun getRide(rideId: String): Ride?
}